### ISSUE TITLE
Calling `setAxisMaxValue` leads to transform error.

### ISSUE THREAD

**Issue by nicholassmith** [danielgindi/Charts#1549]
If I call:

`[self.userDataChartView.leftAxis setAxisMaxValue:100];`

or 

`[self.userDataChartView.xAxis setAxisMaximum:xVal + 1];`

More than once it will lead to:

`<Error>: CGAffineTransformInvert: singular matrix.`

I have a view that allows a user to view multiple datasets by tapping a button, the datasets have different value ranges so I need to be able to specify what the minimum and maximum values should be for the range, but if those values are called more than once it seems like the axis loses the range. 

Have tried:

`[self.userDataChartView.xAxis resetCustomAxisMax];
  [self.userDataChartView.xAxis resetCustomAxisMin];`
 
But they do nothing to resolve the problem.

**Comment by liuxuan30** [comment]
can you try if it's xAxis that giving the error?

**Comment by nicholassmith** [comment]
I tested both in isolation and they both had the same issue, but I solved it by calling:

`[self.userDataCharView autoScale]; [self.userDataChartView prepareValuePxMatrix];`

Not sure which of the two actually resolved the issue, I saw them in the same header and put them in and it fixed the issue. My gut instinct says `prepareValuePxMatrix` because the error is from a singular matrix so it feels logical. Definitely fixed the problem though.
